workflow/executor/resource.go

```diff
@@ -210,6 +211,10 @@ func (we *WorkflowExecutor) checkResourceState(ctx context.Context, selfLink str
 		return true, errors.Errorf(errors.CodeNotFound, "The error is detected to be transient: %v. Retrying...", err)
 	}
 	if err != nil {
+		err = errors.Cause(err)
+		if apierr.IsNotFound(err) {
+			err = errors.Errorf(errors.CodeNotFound, "The resource has been deleted while its status was still being checked. Will not be retried: %v", err)
```

Comment:
maybe return at this line rather than shadowing error?

Reply:
Could you elaborate? It will actually get returned at the line below anyways. Same with other types of errors. We can detecting this particular type of error here to provide a more informative error message.

Reply:
you're going to return in 2 lines, so why wait?

Reply:
Sure, updated.